### DISCUSSION TITLE
Ethshadow: Ethereum simulation with Shadow

### DISCUSSION THREAD

**Discussion by haxxpop** [shadow/shadow#3425] [Show and tell]
Today, I released Ethshadow which is a tool to simulate Ethereum networks using Shadow. See https://github.com/ethereum/ethshadow

I will also present it at Devcon this year. See https://devcon.org. I think Shadow will be exposed more to the Ethereum community afterwards.

Two years ago. No one that I know in the community really knows Shadow. I presented to them that it's a great simulator. After that, it's well known internally among client developers. But today, it's already shown in the Ethereum github which will help gain more visibility to a broader community.

**Comment by stevenengler** [comment]
Cool! I'll take a closer look at it later when I have some time, but it's nice seeing more ways that people find Shadow useful.

Also if there are issues you've had with Shadow or missing features you'd want for the Etherium use case, we'd appreciate any feedback on that (maybe as a new discussion item). None of the core Shadow developers have a lot of time to work on new features at the moment, but we're still doing bug fixes and maintenance. It would be good to have some ideas about what other people want from Shadow outside of just Tor use cases.

**Reply by haxxpop** [reply]
> ). None of the core Shadow developers have a lot of time to work on new features at the moment, but we're still doing bug fixes and maintenance

No worries. If there are missing features, I can send pull requests (just like I did in the past https://github.com/shadow/shadow/pulls?q=is%3Apr+author%3Appopth+is%3Aclosed)
